vault/barrier_aes_gcm.go

```diff
@@ -75,18 +86,40 @@ type AESGCMBarrier struct {
 	// of const to allow for testing
 	currentAESGCMVersionByte byte
 
-	initialized atomic.Bool
+	initialized atomic2.Bool
+
+	randomReaderAccessor func() io.Reader
+	RotationConfig       *configutil.BarrierConfig
 }
 
 // NewAESGCMBarrier is used to construct a new barrier that uses
 // the provided physical backend for storage.
-func NewAESGCMBarrier(physical physical.Backend) (*AESGCMBarrier, error) {
+func NewAESGCMBarrier(physical physical.Backend, rotConfig *configutil.BarrierConfig, randomReaderAccessor func() io.Reader, l log.Logger) (*AESGCMBarrier, error) {
+	// Should we have minimums?
```

Comment:
I don't think we need minimums, do we? We could probably give guidance on what to set these values to.

Reply:
The config param disappears with the api based config, but yeah, I'll add a comment on what the reader can/should be.